components/ModEditFacilityOrProfessionalTopbar.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+    <div class="flex flex-row justify-between w-full">
+        <div>
+            <button
+                data-testid="mod-edit-facility-hp-topbar-copy-id"
+                class="flex flex-row w-90 bg-neutral p-2 m-2 border-2 border-slate-400 rounded hover"
+                @click="copyFacilityOrHPId"
+            >
+                ID: {{ selectedId }}
+                <SVGSuccessCheckMark
+                    v-if="showCopySuccessIcon"
+                    role="img"
+                    title="clipboard copy"
+                    class="ml-2 w-6"
+                />
+                <SVGCopyContent
+                    v-else
+                    role="img"
+                    title="successful copy"
+                    class="ml-2 w-6"
+                />
+            </button>
+        </div>
+        <div class="flex flex-row justify p-2 font-bold ">
+            <button
+                type="button"
+                class="flex justify-center items-center rounded-full bg-secondary-bg border-primary-text-muted
+                border-2 w-28 text-sm mr-2"
+                data-testid="mod-edit-facility-hp-topbar-update"
+            >
+                <span>
+                    {{ $t('modEditFacilityOrHPTopbar.updateAndExit') }}
+                </span>
+            </button>
+            <button
+                type="button"
+                class="flex justify-center items-center rounded-full bg-secondary-bg border-primary border-2 w-28 text-sm mr-2 "
+                data-testid="mod-edit-facility-hp-topbar-delete"
+            >
+                {{
+                    $t('modEditFacilityOrHPTopbar.delete') }}
+            </button>
+        </div>
+    </div>
+</template>
+
+<script setup lang="ts">
+import { computed, ref, type ComputedRef, type Ref } from 'vue'
+import SVGCopyContent from '~/assets/icons/content-copy.svg'
+import SVGSuccessCheckMark from '~/assets/icons/checkmark-square.svg'
+import { useFacilitiesStore } from '~/stores/facilitiesStore'
+import { useHealthcareProfessionalsStore } from '~/stores/healthcareProfessionalsStore'
+import { useModerationScreenStore, ModerationScreen } from '~/stores/moderationScreenStore'
+
+// Initialize the stores in use
+const facilitiesStore = useFacilitiesStore()
+const healthcareProfessionalsStore = useHealthcareProfessionalsStore()
+const moderationScreenStore = useModerationScreenStore()
+
+// Initialize the value of the selected Id based off of Moderation Screen
+const selectedId: ComputedRef<string> = computed(() => setSelectedId())
+
+const showCopySuccessIcon: Ref<boolean> = ref(false)
+
+const copyFacilityOrHPId = async () => {
+    try {
+        await navigator.clipboard.writeText(selectedId.value)
+        showCopySuccessIcon.value = true
+        setTimeout(() => {
+            showCopySuccessIcon.value = false
+        }, 2000)
+    } catch (err: unknown) {
+        console.error(`Failed to copy ID ${selectedId.value}: ${err}`)
```

Comment:
we might want to show it failed to the user here as a toast and not just console